back/src/Model/GameRoom.ts

```diff
@@ -249,32 +246,80 @@ export class GameRoom {
+                        const distance = GameRoom.computeDistanceBetweenPositions(
+                            user.getPosition(),
+                            previewNewGroupPosition
+                        );
+
+                        if (distance > this.groupRadius) {
+                            isOutOfBounds = true;
+                            break;
+                        }
+                    }
+                    group.setOutOfBounds(isOutOfBounds);
                 }
-                const usrPos = foreignUser.getPosition();
-                const grpPos = foreignUser.group.getPosition();
-                const distance = GameRoom.computeDistanceBetweenPositions(usrPos, grpPos);
-
-                if (distance > this.groupRadius) {
-                    if (foreignUser.hasFollowers() || foreignUser.following) {
-                        // If one user is out of the group bounds BUT following, the group still exists... but should be hidden.
-                        // We put it in 'outOfBounds' mode
-                        group.setOutOfBounds(true);
-                        noOneOutOfBounds = false;
-                    } else {
-                        this.leaveGroup(foreignUser);
+            } else if (user.hasFollowers()) {
+                const followingMembers = group.getUsers().filter((currentUser) => currentUser.following === user);
+                let isOutOfBounds = false;
+
+                if (followingMembers.length === group.getUsers().length - 1) {
+                    for (const member of followingMembers) {
+                        const distance = GameRoom.computeDistanceBetweenPositions(
+                            member.getPosition(),
+                            previewNewGroupPosition
+                        );
+
+                        if (distance > this.groupRadius) {
+                            isOutOfBounds = true;
+                            break;
+                        }
+                    }
+
+                    group.setOutOfBounds(isOutOfBounds);
+                }
+            } else {
+                for (const headMember of headMembers) {
+                    if (!headMember.group) {
+                        this.leaveGroup(headMember);
+                        headMembers = group.getGroupHeads();
+                        return;
+                    }
+
+                    const usrPos = headMember.getPosition();
+                    const distance = GameRoom.computeDistanceBetweenPositions(usrPos, previewNewGroupPosition);
+
+                    if (distance > this.groupRadius) {
+                        hasKickOutSomeone = true;
+                        break;
                     }
                 }
-            });
-            if (noOneOutOfBounds && !user.group?.isEmpty()) {
-                group.setOutOfBounds(false);
+
+                if (hasKickOutSomeone) {
+                    this.leaveGroup(user);
+                }
```

Comment:
I don't understand this part. Why would the "current" user (`user`) leave the group if another user (`headMember`) is too far away?

Reply:
If the current moving user has kicked another user from the radius, the moving user leaves the group because he is too far away.